IncrementalPCA.py

```diff
@@ -0,0 +1,155 @@
+import torch.nn as nn
+import torch
+import torch.nn.functional as F
+from PCA import PCA
+
+
+class IncrementalPCA(PCA):
+    """
+    An implementation of Incremental Principal Components Analysis (IPCA) that leverages PyTorch for GPU acceleration.
+
+    This class provides methods to fit the model on data incrementally in batches, and to transform new data
+    based on the principal components learned during the fitting process.
+
+    Attributes:
+        n_components (int, optional): Number of components to keep.
+        n_features (int, optional): Number of original components.
+        mean: The mean of the data, if not given it will be calculated and updated from the batched data.
+    """
+
+    def __init__(self, n_components: int, n_features: int, mean=None):
+        super(IncrementalPCA, self).__init__()
+        assert n_components < n_features
+        self.n_components = n_components
+        self.n_features = n_features
+
+        self.fixed_mean = mean is not None
+
+        self.register_buffer('mean_', mean if self.fixed_mean else torch.zeros(n_features).float())
+        self.register_buffer('components_', torch.zeros((n_components, n_features)).float())
+        self.register_buffer('singular_values_', torch.zeros((n_components,)).float())
+
+        self.register_buffer('N_', torch.tensor([0]))
+
+    def _validate_data(self, X):
+        assert X.shape[0] >= self.n_components
+        assert X.shape[1] >= self.n_components
+        assert X.device == self.N_.device
+
+    def update_mean_var(self, X):
```

Comment:
Looks like this method is doing the job of this function from the original implementation:
- Called here: https://github.com/scikit-learn/scikit-learn/blob/9e38cd00d032f777312e639477f1f52f3ea4b3b7/sklearn/decomposition/_incremental_pca.py#L324
- Definition here: https://github.com/scikit-learn/scikit-learn/blob/9e38cd00d032f777312e639477f1f52f3ea4b3b7/sklearn/utils/extmath.py#L1028
What happened to returning the variance as well? Is it not required?

If this class doesn't returned explained variance it's not required: https://github.com/scikit-learn/scikit-learn/blob/9e38cd00d032f777312e639477f1f52f3ea4b3b7/sklearn/decomposition/_incremental_pca.py#L354

Oh it looks like sklearn computes the mean again, that's inefficient: https://github.com/scikit-learn/scikit-learn/blob/9e38cd00d032f777312e639477f1f52f3ea4b3b7/sklearn/decomposition/_incremental_pca.py#L337
I see why you pass out `new_mean` now.